proposals/NNNN-android-oreo-foreground-services.md

```diff
@@ -0,0 +1,145 @@
+
+![screenshot](../assets/proposals/NNNN-android-oero-foreground-services/sdl_notification.jpg)
+
+[SDL Android v4.6.0](https://github.com/smartdevicelink/sdl_android/tree/release/4.6.0_RC) (currently under development) improves the behavior adding flags to known bluetooth devices. This flag will indicate if the app ever connected and registered to the connected Bluetooth device. 
+
+> The new logic will keep the service in the foreground for 10 seconds if it is a device that has never been seen before or rather the first time the device has connected over bluetooth, if the device does NOT connect, the mac address of the device will be stored with a flag indicating it should be ignored. If it DOES connect over SDL, the notification will then be present for 20 seconds on connection to that device to ensure a connection happens in all scenarios. The router service will still be started for devices that have the flag indicating to ignore those devices, but the service is immediately taken out of the foreground state. This is necessary since in Android O+ services have to be started into the foreground when started by background applications.
+
+This behavior is way better compared to v4.5.0 however the SDL related notification will still be visible to pure non-SDL users. We expect the the risk app partners may weigh benefits integrating SDL and complains from non-SDL users potentially giving bad rates or stop using the app.
+
+## Proposed solution
+
+The proposed solution is to change the behavior how apps prepare the SDL connection to Bluetooth devices. The idea is to create the SDL proxy only while the app is in foreground on the phone in case it never connected to a head unit. It's not allowed to start the router service from the background if the connecting BT device is not known that it supports SDL. Once the app has registered to a head unit it's allowed to start the router service in future connections with that BT device.
+
+As a result no SDL code will cause any visible elements on the Android phone until the app established a connection to the vehicle.
+
+The first time use will be different as the phone must be unlocked and the SDL app must be in foreground. Any subsequent BT connnection to that device will make the app start the service from the background using the Android notification. It's expected that the user may accept the existance of this notification.
+
+Once the first time use is done for the first SDL apps it's router service may notify other apps of the acceptance of SDL usage. This avoid to redo the guid for each app but only for the first app installed on the phone. 

```

Comment:
I believe the first sentence should say"...done for the first SDL app, it's router..." 

Please also update the second sentence to "This avoids the need to redo the guide for each app, and instead only for the first app installed on the phone."